src/main/java/org/wildfly/security/keystore/LdapKeyStore.java

```diff
@@ -102,7 +103,7 @@ public LdapKeyStore build() {
                     filterAlias, filterCertificate, filterIterate, createPath, createRdn, createAttributes, aliasAttribute,
                     certificateAttribute, certificateType, certificateChainAttribute, certificateChainEncoding,
                     keyAttribute, keyType);
-            return new LdapKeyStore(spi, null, null);
+            return new LdapKeyStore(spi, new WildFlyElytronProvider(), "LdapRealm");
```

Comment:
You can still subclass classes with protected constructors (which are visible to subclasses by definition).  In fact for an abstract class, a protected constructor is identical to a public constructor because either way you cannot instantiate abstract classes.